src/modules/form/util/formUtil.ts

```diff
@@ -865,6 +876,14 @@ export const buildAutofillDependencyMap = (
     item.autofillValues.forEach((av) => {
       if (viewOnly && !av.autofillReadonly) return;
 
+      if (av.formula) {
```

Comment:
I agree that "formula" is something that gets evaluated at the "autofill" stage.
I think the `formula`  field should probably be on the `AutofillValue` object, not the `FormItem` object. 
Something like 
```
autofill: [
  {
      value_formula: 'x + y + 17',
      autofill_when: <condition>
   },
   // more autofill conditions can go here. The first one who's "when" condition evalutes to true will be rendered
]
```


This would have a few benefits imo
1. Clearer how the formula is used (because its on an autofill)
2. Easier to expand "formula" for other use-cases later. For example, we might want to evaluate a formula on the `InitialValue` object to specify the initial value for a field. Or, on a `ValueBound` like above. Each of those objects defines a distinct behavior. A "formula" is more of a tool that could be used for each of those behaviors. Wdyt? I'm not sure I am finding the right words to describe it, but there's something there.
3. Autofill formula can be conditionally applied as above


There is one issue with it:
* `autofill_when` is non-nullable, so you need to specify a condition. In this case we always want it to evaluate (I think?). This has been a minor annoyance I am aware of. We should fix it so there is an easy way to specify an autofill value that always runs. (Either by making the condition nullable, or adding way to specify always)

Reply:
>I think the formula field should probably be on the AutofillValue object, not the FormItem object.

Sorry, I'm probably misunderstanding but I think the formula [is already on autofillValue](https://github.com/greenriver/hmis-warehouse/pull/4031/files#diff-8ef987cb4091bd67da4667c6c74ff8ba133bb9e5f5f03ae7cd621f8fcd85564aR20). Maybe we can huddle about it this week; I'm sure there's plenty of room for improvement in my implementation 😅 

> Edit to add: Is the formula aware of "dependencies" from the autofilled item to the other item values that the formula depends on

I think so, or at least it should have the same dependency behavior as sum_questions

Reply:
Whoops 🤦 my bad. Got it